src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -342,46 +931,87 @@ private static String getFirstSentence(DetailNode ast) {
     }
 
     /**
-     * Tests if first sentence contains forbidden summary fragment.
+     * Finds and return if the HTML Format Inline Tag has allowed Inline Tags.
      *
-     * @param firstSentence String with first sentence.
-     * @return true, if first sentence contains forbidden summary fragment.
+     * @param ast Javadoc root Node.
+     * @return true, if first sentence contains allowed tags.
      */
-    private boolean containsForbiddenFragment(String firstSentence) {
-        final String javadocText = JAVADOC_MULTILINE_TO_SINGLELINE_PATTERN
-                .matcher(firstSentence).replaceAll(" ").trim();
-        return forbiddenSummaryFragments.matcher(trimExcessWhitespaces(javadocText)).find();
+    private static boolean validInlineTagInsideHtmlFormat(DetailNode ast) {
+        boolean contains = true;
+        final DetailNode inlineTagNode = getInlineTagNodeInsideHtmlTags(ast);
+        if (inlineTagNode != null) {
+            final DetailNode[] htmlChild = inlineTagNode.getChildren();
+            for (DetailNode node : htmlChild) {
+                if (node.getType() == JavadocTokenTypes.DESCRIPTION) {
+                    contains = checkInlineTagInsideDescription(node);
+                }
+            }
+        }
+        return contains;
     }
 
     /**
-     * Trims the given {@code text} of duplicate whitespaces.
+     * Finds and return if the Inline Tag in description Node has allowed Inline Tags.
      *
-     * @param text The text to transform.
-     * @return The finalized form of the text.
+     * @param description Description Node.
+     * @return true, if first sentence contains allowed tags.
      */
-    private static String trimExcessWhitespaces(String text) {
```

Comment:
this is violation from our Check, we can always suppress violations when required and reasonable.
please restore method and add this SummaryJavadocCheck.java in suppression for now.
We will think on what to do with it at the end.

Reply:
https://github.com/checkstyle/checkstyle/blob/master/config/suppressions.xml

Reply:
Right now it is not required since we removed one method, I will add later if required.